src/class-handler.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * Duplicate Post handler class for duplication actions.
+ *
+ * @package Duplicate_Post
+ * @since 4.0
+ */
+
+namespace Yoast\WP\Duplicate_Post;
+
+/**
+ * Represents the handler for duplication actions.
+ */
+class Handler {
+
+	/**
+	 * Post_Duplicator object
+	 *
+	 * @var Post_Duplicator
+	 */
+	private $post_duplicator;
+
+	/**
+	 * Initializes the main class.
+	 *
+	 * @param Post_Duplicator $post_duplicator The Post_Duplicator object.
+	 */
+	public function __construct( Post_Duplicator $post_duplicator ) {
+		$this->post_duplicator = $post_duplicator;
+		$this->register_hooks();
+	}
+
+	/**
+	 * Adds hooks to integrate with WordPress.
+	 *
+	 * @return void
+	 */
+	private function register_hooks() {
+		\add_action( 'admin_action_duplicate_post_copy_for_rewrite', [ $this, 'rewrite_link_action_handler' ] );
+		\add_action( 'admin_init', [ $this, 'add_bulk_handlers' ] );
+	}
+
+	/**
+	 * Handles the action for copying a post for the Rewrite and Republish feature.
+	 *
+	 * @return void
+	 */
+	public function rewrite_link_action_handler() {
+		if ( ! \duplicate_post_is_current_user_allowed_to_copy() ) {
+			\wp_die( \esc_html__( 'Current user is not allowed to copy posts.', 'duplicate-post' ) );
+		}
+
+		if ( ! ( isset( $_GET['post'] ) || isset( $_POST['post'] ) || // Input var okay.
+			( isset( $_REQUEST['action'] ) && 'duplicate_post_copy_for_rewrite' === $_REQUEST['action'] ) ) ) { // Input var okay.
+			\wp_die( \esc_html__( 'No post to duplicate has been supplied!', 'duplicate-post' ) );
+		}
+
+		$id = ( isset( $_GET['post'] ) ? intval( \wp_unslash( $_GET['post'] ) ) : intval( \wp_unslash( $_POST['post'] ) ) ); // Input var okay.
+
+		\check_admin_referer( 'duplicate-post_rewrite_' . $id ); // Input var okay.
+
+		$post = \get_post( $id );
+
+		if ( ! $post ) {
+			\wp_die(
+				\esc_html(
+					__( 'Copy creation failed, could not find original:', 'duplicate-post' ) . ' '
+					. $id
+				)
+			);
+		}
+
+		if ( $post->post_status !== 'publish' ) {
+			\wp_die(
+				\esc_html(
+					__( 'You cannot create a copy for Rewrite and Republish if the original is not published.', 'duplicate-post' )
+				)
+			);
+		}
+
+		$new_id = $this->post_duplicator->create_duplicate_for_rewrite_and_republish( $post );
+
+		if ( \is_wp_error( $new_id ) ) {
+			\wp_die(
+				\esc_html(
+					__( 'Copy creation failed, could not create a copy.', 'duplicate-post' )
```

Comment:
__( should be backslashed. Question: why not use `esc_html__()`? Also for the previous cases.

Reply:
Will fix the backslashes.
I changed something and I didn't notice that I coud use the more compact syntax. I will fix that, except for the `Copy creation failed, could not find original:` message since it has `$id` appended at the end and I don't want to change the line in the translations